traffic_ops/traffic_ops_golang/server/detail.go

```diff
@@ -247,42 +274,43 @@ JOIN type t ON server.type = t.id
 	}
 	idRows, err := AddWhereClauseAndQuery(tx, fmt.Sprintf(queryFormatString, ""), hostName, physLocationID, orderByStr, limitStr)
 	if err != nil {
-		return nil, errors.New("querying delivery service eligible servers: " + err.Error())
+		return nil, fmt.Errorf("querying delivery service eligible servers: %w", err)
 	}
-	defer idRows.Close()
+	defer log.Close(idRows, "getting IDs for server details names")
 	var serverIDs []int
 	for idRows.Next() {
 		var serverID *int
 		err := idRows.Scan(&serverID)
 		if err != nil {
-			return nil, errors.New("querying delivery service eligible server ids: " + err.Error())
+			return nil, fmt.Errorf("querying delivery service eligible server ids: %w", err)
 		}
 		serverIDs = append(serverIDs, *serverID)
 	}
 	serversMap, err := dbhelpers.GetServersInterfaces(serverIDs, tx)
 	if err != nil {
-		return nil, errors.New("unable to get server interfaces: " + err.Error())
+		return nil, fmt.Errorf("unable to get server interfaces: %w", err)
 	}
 	rows, err := AddWhereClauseAndQuery(tx, fmt.Sprintf(queryFormatString, dataFetchQuery), hostName, physLocationID, orderByStr, limitStr)
 	if err != nil {
-		return nil, errors.New("Error querying detail servers: " + err.Error())
+		return nil, fmt.Errorf("querying detail servers: %w", err)
 	}
 
-	defer rows.Close()
+	defer log.Close(rows, "getting server details data")
 	sIDs := []int{}
 	servers := []tc.ServerDetailV40{}
 
-	serviceAddress := util.StrPtr("")
-	service6Address := util.StrPtr("")
-	serviceGateway := util.StrPtr("")
-	service6Gateway := util.StrPtr("")
-	serviceNetmask := util.StrPtr("")
-	serviceInterface := util.StrPtr("")
-	serviceMtu := util.StrPtr("")
+	serviceAddress := new(string)
```

Comment:
just personal preference, no mechanical reason. For blank strings `new` is technically faster because there's no extra function call stack, but a good compiler probably optimizes that away if it makes sense to do so.